packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2896,3 +3025,404 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// The status of the [SelectionListener].
+enum SelectionListenerStatus {
+  /// Indicates the selection under a [SelectionListener] is changing.
+  changed,
+
+  /// Indicates the selection under a [SelectionListener] is finalized.
+  finalized,
+}
+
+/// Signature for listeners attached using [SelectionListenerNotifier.addStatusListener].
+typedef SelectionListenerStatusCallback = void Function(SelectionListenerStatus status);
+
+/// Notifies its listeners of changes in selection.
+class _SelectableRegionSelectionStatusNotifier extends ChangeNotifier {
+  SelectionListenerStatus _selectionListenerStatus = SelectionListenerStatus.finalized;
+  SelectionListenerStatus get selectionListenerStatus => _selectionListenerStatus;
+  set selectionListenerStatus(SelectionListenerStatus newStatus) {
+    _selectionListenerStatus = newStatus;
+    notifyListeners();
+  }
+}
+
+/// Notifies its listeners when the selection under a [SelectableRegion] or
+/// [SelectionArea] is finalized or has changed.
+class _SelectableRegionScope extends InheritedNotifier<_SelectableRegionSelectionStatusNotifier> {
```

Comment:
One thing about this approach is that someone outside the framework or this file, can't be notified of these changes (the closest they can get is hooking into `dispatchSelectionEvent`) at the selection delegate level.

Reply:
You mean the notifier will be notified when there're changes in the region, including changes that happen outside the subtree the `SelectionListener` is in right? Does that mean the status listening API should live somewhere else since it's not bounded by the `SelectionListener`'s scope? 

The current documentation of `addStatusListener`:
  /// Calls the listener every time the [SelectionListenerStatus] of a [SelectionListener]
  /// is updated.
  ///
  /// Listeners can be removed with [removeStatusListener].

That seems to indicate the status listeners have the same scope as regular listeners?

Reply:
Talked offline and updated the documentation to reflect the notification is coming from [SelectableRegion].

Reply:
@chunhtai do you have any thoughts on this? The `SelectionListener` and `SelectableRegion` become coupled with this approach, where before any generic selection delegate could listen to events through `dispatchSelectionEvent`, now there is a private notifier. Also if someone else other than `SelectableRegion` dispatches selection events down the subtree the `SelectionListener` would miss these since it only listens to `SelectableRegion`.

Reply:
do you mean someone that call dispatchSelectionEvent above or under the selectableRegion directly? I think that should be fine right?

the selection change should still bubble up by from selectionGeometry somehow right?

Reply:
I think there are two cases here, people that dispatchSelectionEvent under a selectableRegion and people that would like to listen to selection changes like we are.

1. People that dispatchSelectionEvent under a selectableRegion will not have their events caught by `SelectionListener`. I updated `SelectionListenerNotifier` to notify when events change `SelectionGeometry` through `dispatchSelectionEvent`, users can use `SelectableRegionScope` to listen to updates from `SelectableRegion`.
2. People that would like to listen to `SelectableRegionStatusNotifier` like we are internally with `SelectionListener` cannot. I updated the implementation to make `SelectableRegionScope` public so users can now do this.